reactor-pool/src/main/java/reactor/pool/AbstractPool.java

```diff
@@ -409,6 +412,8 @@ Context currentContext() {
 		@Override
 		public void run() {
 			if (Borrower.this.compareAndSet(false, true)) {
+				// this is failure, a timeout was observed
+				pool.metricsRecorder.recordPendingFailureAndLatency(pool.clock.millis() - pendingAcquireStart);
```

Comment:
```suggestion
				stopPendingCountdown(false);
```

Reply:
Otherwise, the `timeoutTask` is not disposed

Reply:
nice catch, thanks